core/extension/native_extension.cpp

```diff
@@ -253,6 +254,10 @@ void NativeExtension::_unregister_extension_class(const GDNativeExtensionClassLi
 	if (ext->native_extension.parent != nullptr) {
 		ext->native_extension.parent->children.erase(&ext->native_extension);
 	}
+	for (MethodBind *E : ext->native_extension.methods) {
+		memdelete(E);
+	}
+	ext->native_extension.methods.clear();
```

Comment:
I'm not sure that's the right way to fix it. The method pointer is passed to ClassDB with `ClassDB::bind_method_custom(class_name, method);` which stores it in `        type->method_map[p_method->get_name()] = p_method;`.

So IMO it's ClassDB that should delete it and remove it from its `method_map`, probably when calling `unregister_extension_class`.

Reply:
True, it is responsible for deleting it normally when `ClassDB::cleanup()` is called. Fixed